### DISCUSSION TITLE
Media Channel Support

### DISCUSSION THREAD

**Discussion by Panthr75** [discord-net/Discord.Net#2721] [General]
I'm working on media channel support for my [custom fork of Discord.NET](https://github.com/Panthr75/Based-Discord.Net) to be used inside my own library. Once the pull request is finished however, I would like to make a pull request here.

The current pull request can be viewed [here](https://github.com/Panthr75/Based-Discord.Net/pull/1), and I want to discuss some of the changes here.

**Comment by Panthr75** [comment]
One big question I have is if I should separate out some common properties of `IForumChannel` and `IMediaChannel` to a respective `IThreadContainerChannel`, or something of the sorts, or should I keep them the same.

**Reply by Panthr75** [reply]
I've decided on `IDiscussionChannel`

**Comment by Misha-133** [comment]
From what I've seen in docs - it looks like there's no need to make a common interface, rather the `IMediaChannel` can be inherited from `IForumChannel`

**Reply by Panthr75** [reply]
There are a couple of differences. Apparently you cant use webhooks in newly created media channels, and there's no layout you can set, and apparently you can't mark them as nsfw. I have to check all of these tho

**Reply by Misha-133** [reply]
> you cant use webhooks in newly created media channels 
yuh, and the API returns `Unknown channel`, so I'm not sure if this is intended behavior or a bug.
> and there's no layout you can set
Yeah, it's locked to grid layout
> can't mark them as nsfw
Nope, you definitely can.

**Comment by Misha-133** [comment]
Also, just to note: I have a local branch with the media channel support implemented, just've been waiting for a way to test it

**Reply by Panthr75** [reply]
Same honestly

**Reply by Panthr75** [reply]
One of my friends servers has a media channel, might kindly ask for permission to test lol

**Reply by Misha-133** [reply]
Yeah, I got a channel for testing so it's not a problem anymore... I'll draft PR my branch then

**Reply by Panthr75** [reply]
Can you send a link of that pr to update my pr?

**Reply by Misha-133** [reply]
https://github.com/discord-net/Discord.Net/pull/2725

**Reply by Panthr75** [reply]
Updated my PR with yours, with some comments as well.
